feature-gate/program/src/instruction.rs

```diff
@@ -0,0 +1,87 @@
+//! Program instructions
+
+use {
+    borsh::{BorshDeserialize, BorshSchema, BorshSerialize},
+    solana_program::{
+        instruction::{AccountMeta, Instruction},
+        pubkey::Pubkey,
+        system_program,
+    },
+};
+
+/// Feature Gate program instructions
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, BorshSchema, PartialEq)]
+pub enum FeatureGateInstruction {
+    /// Submit a feature for activation.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[ws]` Feature account (must be a system account)
+    ///   1. `[s]` Authority
```

Comment:
What's this supposed to do?

Reply:
I'm basically giving the program the ability to do [this](https://github.com/solana-labs/solana/blob/master/sdk/program/src/feature.rs#L53-L57)

Reply:
Sorry, I meant the `authority` specifically

Reply:
Prob going to remove as a result of your other comment below. I was under the impression there was some authority additional to the feature keypair involved. I get the picture now I think